### PR TITLE
FEAT: Raptor x simluation setup

### PR DESCRIPTION
**Pull request by svandenb-dev**
RaptorX simulatiom setup support added. Works only with Release 2024.2 and higher.

### CODE REVIEW

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+        self.settings.enabled = value
+
+    @property
+    def position(self):
+        return self._edb_setup_info.Position
+
+    @position.setter
+    def position(self, value):
+        if isinstance(value, int):
+            self._edb_setup_info.Position = value
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:71]: Can you log an error if value is not int ?

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+        return self._advanced_settings
+
+    @property
+    def enabled(self):
+        return self._simulation_settings.Enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        if isinstance(value, bool):
+            self._simulation_settings.Enabled = value
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:131]: Can you log an error if value is not bool ?

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+
+    @property
+    def net_settings_options(self):
+        """A list of Name, Value pairs that stores advanced option."""
+        return [val for val in list(self._advanced_settings.NetSettingsOptions)]
+
+    @net_settings_options.setter
+    def net_settings_options(self, value):
+        if isinstance(value, list):
+            self._advanced_settings.NetSettingsOptions = convert_py_list_to_net_list(value)
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:232]: Can you log an error if value is not list ?

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+    def use_accelerate_via_extraction(self):
+        """Setting this option will simplify/merge neighboring vias before sending the layout for processing
+        to the mesh engine and to the EM engine.
+        """
+        return self._advanced_settings.UseAccelerateViaExtraction
+
+    @use_accelerate_via_extraction.setter
+    def use_accelerate_via_extraction(self, value):
+        if isinstance(value, bool):
+            self._advanced_settings.UseAccelerateViaExtraction = value
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:269]: Can you log an error if value is not bool ?

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+    def use_cells_per_wavelength(self):
+        """This setting describes the number of cells that fit under each wavelength. The wavelength is calculated
+        according to the Max Frequency or the Mesh Frequency, unless specified by user through this setting.
+        """
+        return self._advanced_settings.UseCellsPerWavelength
+
+    @use_cells_per_wavelength.setter
+    def use_cells_per_wavelength(self, value):
+        if isinstance(value, bool):
+            self._advanced_settings.UseCellsPerWavelength = value
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:291]: Can you log an error if value is not bool ?

File: src/pyedb/dotnet/edb_core/utilities/simulation_setup.py [hunk 6]
```diff
@@ -122,7 +126,7 @@ def enabled(self):
 
     @enabled.setter
     def enabled(self, value):
-        edb_setup_info = self.get_sim_setup_info
+        # edb_setup_info = self.get_sim_setup_info
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/utilities/simulation_setup.py:129]:
```suggestion
```
Should this me uncommented or removed ?

File: src/pyedb/dotnet/edb_core/utilities/simulation_setup.py [hunk 7]
```diff
@@ -722,6 +730,9 @@ def set_frequencies(self, frequency_list=None, update=True):
                 ["linear scale", "0.1GHz", "10GHz", "0.1GHz"],
             ]
         temp = []
+        if isinstance(frequency_list, list):
+            if not isinstance(frequency_list[0], list):
+                frequency_list = [frequency_list]
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/utilities/simulation_setup.py:735]:
```suggestion
        if isinstance(frequency_list, list) and not isinstance(frequency_list[0], list):
            frequency_list = [frequency_list]
```

File: tests/legacy/system/test_edb.py [hunk 8]
```diff
@@ -1732,3 +1730,91 @@ def test_multizone(self, edb_examples):
         assert defined_ports
         assert project_connexions
         edbapp.close_edb()
+
+    @pytest.mark.skipif(not desktop_version == "2024.2", reason="Only supported with 2024.2 and higher")
```
Comment (SMoraisAnsys) [inline tests/legacy/system/test_edb.py:1734]:
```suggestion
    @pytest.mark.skipif(not desktop_version == "2024.2", reason="Only supported with 2024.2 and higher")
```
Here you won't cover 2025.1, please update the skipif.

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
Nice job here, I just have a few remarks. I stopped half way with the if isinstance(...) remarks, there are some that I did not report.
About the docstrings, you might want to make them uniform because right now it feels like some times you speak to a user, other times a developer, and other times an efficient quick description (which i prefer tbh)

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py (hunk 5 above)
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:317]: done

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py [hunk 9]
```diff
@@ -0,0 +1,448 @@
...
+
+    @property
+    def use_auto_removal_sliver_poly(self):
+        """Setting this option simplifies layouts by aligning slightly misaligned overlapping polygons."""
+        return self._advanced_settings.UseAutoRemovalSliverPoly
+
+    @use_auto_removal_sliver_poly.setter
+    def use_auto_removal_sliver_poly(self, value):
+        if isinstance(value, bool):
+            self._advanced_settings.UseAutoRemovalSliverPoly = value
```
[the remark this answers is not in the document]
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:301]: done

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py (hunk 3 above)
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:246]: done

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py (hunk 1 above)
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:73]: done

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py (hunk 2 above)
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:137]: Done

File: src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py (hunk 4 above)
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/edb_data/raptor_x_simulation_setup_data.py:287]: Done

File: tests/legacy/system/test_edb.py (hunk 8 above)
Comment (svandenb-dev) [reply tests/legacy/system/test_edb.py]: Done

File: src/pyedb/dotnet/edb_core/utilities/simulation_setup.py [hunk 10]
```diff
@@ -111,7 +115,7 @@ def _update_setup(self):
         if self._name in self._pedb.setups:
             self._pedb.layout.cell.DeleteSimulationSetup(self._name)
         if not self._pedb.layout.cell.AddSimulationSetup(self._edb_object):
-            raise Exception("Updating setup {} failed.".format(self._name))
+            self._pedb.logger.error("Updating setup {} failed.".format(self._name))
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/utilities/simulation_setup.py:118]: Why don't you want to raise an exception here ?

File: src/pyedb/dotnet/edb_core/utilities/simulation_setup.py [hunk 11]
```diff
@@ -197,14 +204,15 @@ def delete_frequency_sweep(self, sweep_data):
...
-                fsweep.append(val)
-        self.get_sim_setup_info.SweepDataList.Clear()
-        for i in fsweep:
-            self.get_sim_setup_info.SweepDataList.Add(i._edb_object)
-        self._update_setup()
-        return True if name in self.frequency_sweeps else False
+        if self.frequency_sweeps:
+            for k, val in self.frequency_sweeps.items():
+                if not k == name:
+                    fsweep.append(val)
```
[the remark this answers is not in the document]
Comment (SMoraisAnsys) [reply src/pyedb/dotnet/edb_core/utilities/simulation_setup.py:209]: This was not taken into account so i unresolved it.

File: src/pyedb/dotnet/edb_core/utilities/simulation_setup.py (hunk 7 above)
Comment (SMoraisAnsys) [reply src/pyedb/dotnet/edb_core/utilities/simulation_setup.py:734]: This was not taken into account so i unresolved it.

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
Could you add log errors to other setter with isisntance pattern ?

File: tests/legacy/system/test_edb.py [hunk 12]
```diff
@@ -1732,3 +1730,94 @@ def test_multizone(self, edb_examples):
...
+        self.local_scratch.copyfolder(source_path, target_path)
+        edbapp = Edb(edbpath=target_path, edbversion=desktop_version)
+        setup = edbapp.create_raptorx_setup("test")
+        assert "test" in edbapp.setups
+        setup.add_frequency_sweep(frequency_sweep=["linear scale", "0.1GHz", "10GHz", "0.1GHz"])
+        setup.enabled = False
+        assert not setup.enabled
+        assert len(setup.frequency_sweeps) == 1
+        general_settings = setup.settings.general_settings
+        assert general_settings.global_temperature == 22.0
```
Comment (ring630) [inline tests/legacy/system/test_edb.py:1750]: Can you change all property assertion as below? This makes sure all changes are written into edb.
assert edbapp.setups["test"].settings.general_settings-global_temperature == 22.0

File: tests/legacy/system/conftest.py [hunk 13]
```diff
@@ -36,7 +36,7 @@
 example_models_path = os.path.join(dirname(dirname(dirname(os.path.realpath(__file__)))), "example_models")
 
 # Initialize default desktop configuration
-desktop_version = "2024.1"
+desktop_version = "2024.2"
```
Comment (ring630) [inline tests/legacy/system/conftest.py:39]: Do you want to upgrade all test to 2024.2?

Comment (ring630) [inline tests/legacy/system/conftest.py:39]:
```suggestion
desktop_version = "2024.1"
```

File: tests/legacy/system/test_edb.py [hunk 14]
```diff
@@ -1732,3 +1730,94 @@ def test_multizone(self, edb_examples):
...
+        target_path = os.path.join(self.local_scratch.path, "test_raptorx_setup", "test.aedb")
+        self.local_scratch.copyfolder(source_path, target_path)
+        edbapp = Edb(edbpath=target_path, edbversion=desktop_version)
+        setup = edbapp.create_raptorx_setup("test")
+        assert "test" in edbapp.setups
+        setup.add_frequency_sweep(frequency_sweep=["linear scale", "0.1GHz", "10GHz", "0.1GHz"])
+        setup.enabled = False
+        assert not setup.enabled
+        assert len(setup.frequency_sweeps) == 1
+        general_settings = setup.settings.general_settings
```
Comment (ring630) [inline tests/legacy/system/test_edb.py:1749]:
```suggestion
        general_settings = edbapp.setups["test"].settings.general_settings
```

File: tests/legacy/system/test_edb.py [hunk 15]
```diff
@@ -1732,3 +1730,94 @@ def test_multizone(self, edb_examples):
...
+        assert general_settings.global_temperature == 22.0
+        general_settings.global_temperature = 35.0
+        assert general_settings.global_temperature == 35.0
```
Comment (ring630) [inline tests/legacy/system/test_edb.py:1752]:
```suggestion
        assert edbapp.setups["test"].settings.general_settings.global_temperature == 35.0
```

File: src/pyedb/dotnet/edb_core/utilities/simulation_setup.py (hunk 10 above)
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/utilities/simulation_setup.py:118]:
```suggestion
            self._pedb.logger.error("Updating setup {} failed.".format(self._name))
            raise Exception("Updating setup {} failed.".format(self._name))
```

**Review by SMoraisAnsys — DISMISSED** [review]
LGTM

**Pull request merged**